lib/fopen.c

```diff
@@ -56,13 +56,13 @@ CURLcode Curl_fopen(struct Curl_easy *data, const char *filename,
   int fd = -1;
   *tempname = NULL;
 
-  if(stat(filename, &sb) == -1 || !S_ISREG(sb.st_mode)) {
-    /* a non-regular file, fallback to direct fopen() */
-    *fh = fopen(filename, FOPEN_WRITETEXT);
-    if(*fh)
-      return CURLE_OK;
+  *fh = fopen(filename, FOPEN_WRITETEXT);
+  if (!*fh)
     goto fail;
-  }
+  if (fstat(fileno(*fh), &sb) == -1 || !S_ISREG(sb.st_mode))
```

Comment:
`make checksrc` will tell you that we don't use spaces after `if`...

Reply:
My apologies for the oversight, fixed in next commit.